tests/sparseml/pytorch/recipe_template/test_main.py

```diff
@@ -147,3 +163,47 @@ def test_one_shot_applies_sparsification(pruning, quantization, quant_expected,
         ) / sum(torch.numel(weight) for weight in weights)
 
         assert sparsity > 0.75
+
+
+@pytest.mark.parametrize(
+    "num_epochs, init_lr, final_lr, sparsity, lr_func",
+    [
+        (20, 0.001, 0.0, 0.8, "linear"),
+        (2, 0.0001, 0.0, 0.9, "cyclic_linear"),
+    ],
+)
```

Comment:
Optional, but I think it'd be nice to hardcode the dynamically calculated expected values rather than copying the logic from the source code. That way we know we have the correct logic for the outcomes we're expecting. e.g. have expected`num_qat_epochs` and `num_pruning_active_epochs` hardcoded in the test parametrization

Reply:
This is a good suggestion, thank you!